src/core.c/CompUnit/Repository/FileSystem.pm6

```diff
@@ -222,7 +222,7 @@ class CompUnit::Repository::FileSystem
     }
 
     proto method files(|) {*}
-    multi method files($file, Str:D :$name!, :$auth, :$ver, :$api) {
+    multi method files($file, Str:D :$name!, :$auth, :$ver, :$api, :$dist) {
```

Comment:
The dist identity is already declared by the previous parameters. Passing in a dist alongside them doesn't really make fit the api.

Reply:
It's rather about bad naming of the parameter. It's a boolean flag, request for distribution to be returned instead of meta. Your confusion is a clear sign of it needing a rename, but the only option I can come up with is `:$as-dist` and it doesn't appeal to me.

Reply:
:$return-as-dist ?

Reply:
> :$return-as-dist ?

Would like a short version. After all, I expect the argument to be used more often than omitted. The only reason it can't be the default for now is backward compatibility.

Reply:
Can't we make it default depending on a flag passed to CURFS ?

Reply:
> Can't we make it default depending on a flag passed to CURFS ?

We can, but that's another complication I would rather avoid.

Instead, I think I have a more generic solution for language-specific behavior of methods. Based on `PseudoStash` experience, the following works well for 6.e:

```
my class PseudoStash is CORE::v6c::PseudoStash {
```

So, any method in 6.c implementation can do `self.^language_revision eq 'e' ?? do { 6e-ish } !! do { 6c-ish }`.

But that's something for later.

Reply:
You can get the dist using the meta data returned from files via something like `my %file-meta = CUR.files(...); CUR.candidates($file-meta<name>, $file-meta<ver>, ...).head`.

Reply:
Addressed with 672a9b505e2488481343759730b43e224e4a8390. As I still don't see what name would serve better than `:dist` the commit just adds `Bool` type constraint to the parameter.

Reply:
It might be better to create a new method for that, and make this files method deprecated or implementation detail. The new method would be infinitely more useful as a public api than `files`, and the only reason `files` still exists is because its fast pathing with that `<source>` field was needed by `CUR.run-script` to maintain the last overhead.

Reply:
+ method `distributions`, - method `files` (deprecate, surely)?

What is the exact meaning of `source`? Can dump it altogether?

Reply:
Addressed with d6b4b5f160dc3e060d2d9beb510a986362bdc88e.

`source` is used in a couple of locations, so for now I don't touch it.

Reply:
And as to deprecation of `.files`, it is currently rather problematic because it is used by roast's S11-repository/cur-candidates.t. It doesn't feel OK to just replace the method in there.

Reply:
Not to nit pick the name further, but it seems odd that a method called `candidates` takes all those same parts -- except a path name instead of a name -- and does basically the same thing as a method called `distributions`. Ideally they'd probably both be the same method, but there isn't a way to tell the difference beween a path `foo` and a namespace/provides `foo` unfortunately. And adding a named parameter to `candidates`. Additionally `CURFS` already has a method `distribution`, so it also seems a bit off that the plural is a querying mechanism (and only then a specific kind of query). `name-candidates` + `path-candidates` might have been a more cohesive way of naming these originally. That being said I don't have a better name suggestion, so these are just comments.

Reply:
Mostly unrelated to this PR but related to my previous comment is that I always felt `CURFS.distribution` and `CURI.installed` should instead both be called `.distributions` so one can do `$*REPO.repo-chain.map(*.?distributions)` since `.installed` doesn't make sense for a `CURFS` and a singular `.distribution` doesn't make sense for a CURI

Reply:
Why not `.candidates(:$file, ...)`? I like the idea of merging these two into one.

Reply:
As long as both `$name` and `:$file` can't supplied at the same time that could work

Reply:
Done. So, it is now `candidates($name, ...)` and `candidates(:$file!, ...)`. There was an idea of making use of both positional and `:file` an error, but since the named would currently be silently swallowed, I decided not to break this behavior.